src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/multi-cluster/multi-cluster-form/multi-cluster-form.component.ts

```diff
@@ -214,68 +234,42 @@ export class MultiClusterFormComponent implements OnInit, OnDestroy {
             ssl_certificate
           )
           .subscribe({
-            error: () => {
-              this.remoteClusterForm.setErrors({ cdSubmitButton: true });
+            error: (error: any) => {
+              if (error.error.code === 'connection_refused') {
+                this.connectionVerified = false;
+                this.showCrossOriginError = true;
+                this.connectionMessage = error.error.detail;
+                this.crossOriginCmd = `ceph config set mgr mgr/dashboard/cross_origin_url ${window.location.origin} `;
+                this.remoteClusterForm.setErrors({ cdSubmitButton: true });
+                this.notificationService.show(
+                  NotificationType.error,
+                  $localize`Connection to the cluster failed`
+                );
+              } else {
+                this.connectionVerified = false;
+                this.connectionMessage = error.error.detail;
+                this.remoteClusterForm.setErrors({ cdSubmitButton: true });
+                this.notificationService.show(
+                  NotificationType.error,
+                  $localize`Connection to the cluster failed`
+                );
```

Comment:
i see the same code above which means that this can be generalized function. don't duplicate codes, because it'll be difficult for future maintainers to maintain these.

Reply:
Updated the code